packages/next/src/withTRPC.tsx

```diff
@@ -94,31 +95,32 @@ export function withTRPC<
         trpc?: TRPCPrepassProps;
       },
     ) => {
-      const [{ queryClient, trpcClient, ssrState, ssrContext }] = useState(
-        () => {
-          if (props.trpc) {
-            return props.trpc;
-          }
+      const [prepassProps] = useState(() => {
+        if (props.trpc) {
+          return props.trpc;
+        }
 
-          const config = getClientConfig({});
-          const queryClient = new QueryClient(config.queryClientConfig);
-          const trpcClient = trpc.createClient(config);
-          return {
-            queryClient,
-            trpcClient,
-            ssrState: opts.ssr ? ('mounting' as const) : (false as const),
-            ssrContext: null,
-          };
-        },
-      );
+        const config = getClientConfig({});
+        const queryClient = new QueryClient(config.queryClientConfig);
+        const trpcClient = trpc.createClient(config);
+        return {
+          abortOnUnmount: config.abortOnUnmount,
+          queryClient,
+          trpcClient,
+          ssrState: opts.ssr ? ('mounting' as const) : (false as const),
+          ssrContext: null,
+        };
+      });
 
+      const { queryClient, trpcClient, ssrState, ssrContext } = prepassProps;
```

Comment:
couldn't destructure `abortOnUnmount` above since it wasn't a common property in the union. move it down here so i could cast below (not the best, open to suggestions)

Reply:
you could add `abortOnUnmount?: never` to the other side of the union, which would be effectively the same as it not being present, but it would allow for destructuring it...